### DISCUSSION TITLE
There is no way to save part of PDF to PNG with PyPDF2?

### DISCUSSION THREAD

**Discussion by gosforth** [py-pdf/pypdf#936] [Q&A] [answered]
I search for some reliable library to extract part of PDF to PNG.
I think this is not possible directly with PyPDF2, right? I have to use some other library?

Anyway, how can I crop part of PDF; page width and 100 pixels down from top in PyPDF2?

Regards

**Comment by MartinThoma** [comment, marked as answer]
PyPDF2 can extract all images from a PDF.

PyPDF2 currently cannot render the page to an image.

> Anyway, how can I crop part of PDF; page width and 100 pixels down from top in PyPDF2?

You can change the cropbox of a page: https://pypdf2.readthedocs.io/en/latest/user/cropping-and-transforming.html

```
cropbox = page.cropbox
page.cropbox = RectangleObject([cropbox.left, cropbox.bottom, cropbox.right - 100, cropbox.top - 100])
```

However, if I remember it correctly, the unit is not (necessarily) in pixels. I would need to look that up in the specs.

**Reply by gosforth** [reply]
Thank you very much for help!
What would you recommend to save cropped part of PDF to PNG? pdf2image?

**Reply by Butterfly** [reply]
> Thank you very much for help! What would you recommend to save cropped part of PDF to PNG? pdf2image?

Greetings! I have a work-around that I just test drove. `gimp` lets you import PDF files that you can edit and then export as PNG. My test case was a deliberately chosen one-page printscreen from a bill paid yesterday. I hit `Image > Canvas Size` and shrunk it down to 100 pixels tall. It works! And might also help short track looking for how to implement in PyPDF2, to boot. :partying_face:

**Reply by gosforth** [reply]
I'd rather use python - this is background script not hand-mouse activity.
